RUFAS/input_manager.py

```diff
@@ -49,3 +51,39 @@ def _load_metadata(self, metadata_path: str = "input/example_metadata.json") ->
                 om.add_log("load_metadata_success", f"Successfully loaded metadata from {metadata_path}", info_map)
         except Exception as e:
             raise e
+
+    def _load_data(self) -> None:
+        """Loads data from JSON or CSV fileRaises
+
+        Raises
+        ------
+        Exception
+            If an error occurs while opening or reading a data file.
+
+        """
+        files_details = self.__metadata["files"]
+        path_key = "path"
+        info_map = {"class": self.__class__.__name__,
+                    "function": self._load_data.__name__,
+                    }
+        for key, details in files_details.items():
+            file_path = details[path_key]
+            om.add_log("load_data_attempt", f"Attempting to load data for {key} from {file_path}.", info_map)
+            try:
+                if details["type"] == "json":
+                    with open(file_path) as json_file:
+                        data = json.load(json_file)
+                        om.add_log("load_data_successful", f"Successfully loaded data for {key} from {file_path}.",
+                                   info_map)
+                        self.__pool[key] = data
```

Comment:
```suggestion
                        self.__pool[key] = data
                        om.add_log("load_data_successful", f"Successfully loaded data for {key} from {file_path}.",
                                   info_map)
```
Not a big deal but better safe than sorry.

Reply:
I was wondering if we'd wanted to keep that separate in case there was some issue with adding the data to the pool separate from just loading the data but can certainly go along with this suggestion.